index.js

```diff
@@ -107,8 +128,8 @@ app.get('/keep_alive/', logRequest, function (req, res) {
     res.sendStatus(200);
 });
 
-app.post('/post/', logRequest, authorizeUser, checkSingleFileUpload, checkMessagePathQueryParameter, checkTopicQueryParameter, ensureTopicSpecified, function (req, res) {
-    mqttClient.publish(req.body['topic'], req.body['message']);
+app.post('/post/', logRequest, authorizeUser, checkSingleFileUpload, checkMessagePathQueryParameter, checkTopicQueryParameter,checkQoSParameter,checkRetainFlag, ensureTopicSpecified, function (req, res) {
+    mqttClient.publish(req.body['topic'], req.body['message'],req.body['qos'],req.body['retain']);
```

Comment:
Not sure that it will work. Please take a look at https://github.com/mqttjs/MQTT.js/blob/46dc947b630c77322b63a97bf76f43cb15c6d628/examples/wss/client.js#L41

Reply:
Also please format.

Reply:
Looking at https://github.com/mqttjs/MQTT.js/blob/46dc947b630c77322b63a97bf76f43cb15c6d628/examples/wss/client.js#L41, it looks like line 132 should be changed to:

```javascript
let options = {
  qos : req.body.qos,
  retain : req.body.retain == true // force to boolean value
}
mqttClient.publish(req.body['topic'], req.body['message'], options);
```

notice

Reply:
I think this : "retain : req.body.retain == true" can even be shorter like this "retain : !!req.body.retain".

Reply:
True.  (or should I say `!!1` ? )

Reply:
:-)